File: qcue_app/android/app/src/main/kotlin/cn/qcue/app/QcueChannels.kt
package cn.qcue.app

/**
 * QCue S5-R3 — the channel-name + schema-version constants shared by the Android
 * native handlers. These MUST match the Dart `QcueChannels` (lib/core/native/
 * channels.dart) and the iOS `QcueChannels` exactly.
 */
object QcueChannels {
    const val SCHEMA_VERSION = 1

    const val STT = "qcue/stt"
    const val STT_EVENTS = "qcue/stt/events"
    const val SECURE = "qcue/secure"

    const val SHARE = "qcue/share"
    const val SHARE_EVENTS = "qcue/share/events"
    const val WIDGET = "qcue/widget"
    const val WIDGET_EVENTS = "qcue/widget/events"
    const val NOTIF = "qcue/notif"
    const val NOTIF_EVENTS = "qcue/notif/events"
    const val BACKGROUND = "qcue/background"
    const val INSTALLER = "qcue/installer"
}

File: qcue_app/ios/Runner/Native/QcueChannels.swift
import Flutter
import Foundation

/// QCue S5-R3 — channel-name + schema-version constants shared by the iOS native
/// handlers. These MUST match the Dart `QcueChannels` (lib/core/native/channels.dart)
/// and the Android `QcueChannels` exactly.
enum QcueChannels {
    static let schemaVersion = 1

    static let stt = "qcue/stt"
    static let sttEvents = "qcue/stt/events"
    static let secure = "qcue/secure"

    static let share = "qcue/share"
    static let shareEvents = "qcue/share/events"
    static let widget = "qcue/widget"
    static let widgetEvents = "qcue/widget/events"
    static let notif = "qcue/notif"
    static let notifEvents = "qcue/notif/events"
    static let background = "qcue/background"

    /// The shared App Group container id (Share Extension + WidgetKit + the app).
    /// MUST match the App Group entitlement on every target.
    static let appGroup = "group.cn.qcue.shared"

    /// S5-R3 — verify the inbound payload carries the supported major version.
    static func versionOk(_ args: Any?) -> Bool {
        guard let map = args as? [String: Any] else { return false }
        return (map["schemaVersion"] as? Int) == schemaVersion
    }

    static func versionError() -> FlutterError {
        FlutterError(code: "versionMismatch", message: "unsupported schemaVersion",
                     details: ["kind": "versionMismatch", "retryable": false])
    }
}
